pkg/services/vimmachine_test.go

```diff
@@ -188,8 +188,8 @@ var _ = Describe("VimMachineService_GenerateOverrideFunc", func() {
 
 var _ = Describe("VimMachineService_GetHostInfo", func() {
 	var (
-		controllerCtx     *context.ControllerContext
-		machineCtx        *context.VIMMachineContext
+		controllerCtx     *capvcontext.ControllerContext
```

Comment:
```suggestion
		ctx               = context.Background()
		controllerCtx     *capvcontext.ControllerContext
```

To more or less match the previous behaviour: I think we should create a context per top-level `ginkgo.Describe` nad not use the test-package variable `ctx` (which is used for `BeforeEach` and `AfterEach`). But I'm not completely sure and using the test-package variable may also be good (so the context gets canceled on an error).

Or should we use (as in this PR currently) the globally stored context?

What do you think @killianmuldoon and @sbueringer?

Reply:
Although the prow test looks stable now, I think it's better tor revisit the test and setup separate context for different test file or even test case based on specific scenario,  instead of using the test-package variable. This helps ensure that test case is kind of isolated and independent from others, which can make predictable and reliable test results.

Regarding to context cleanup, we could take care of the context cancelation explicitly if needed. 

We could add it as a new task of #2295 , as this PR is already a large one, but I'm fine to update them in this PR.

@chrischdi @sbueringer @killianmuldoon WDYT?

Reply:
lets add it to #2295 👍